Support for Hoodi testnet

gwenall opened the pull request:
Ref: https://github.com/ChorusOne/eth-staking-smith/issues/61

[Hoodi](https://github.com/eth-clients/hoodi) will replace [Holešky](https://github.com/eth-clients/holesky) as a staking, infrastructure and protocol-developer testnet.

gwenall commented:
Caveat: it's depending on https://github.com/ChorusOne/lighthouse fork of sigp/lighthouse that need to have this PR merged : https://github.com/sigp/lighthouse/pull/7145/files

mariavarvaroi commented:
> Caveat: it's depending on https://github.com/ChorusOne/lighthouse fork of sigp/lighthouse that need to have this PR merged : https://github.com/sigp/lighthouse/pull/7145/files

Once the lighthouse pr is merged, please bump the `eth2_network_config` dependency. Please update test cases to verify the new network support (consider replacing instances of `holesky` with `hoodi`)

jenpaff commented:
> > Caveat: it's depending on https://github.com/ChorusOne/lighthouse fork of sigp/lighthouse that need to have this PR merged : https://github.com/sigp/lighthouse/pull/7145/files
> 
> Once the lighthouse pr is merged, please bump the `eth2_network_config` dependency. Please update test cases to verify the new network support (consider replacing instances of `holesky` with `hoodi`)

hoodi won't replace holesky, so fine to just keep it as an additional network.

gwenall commented:
> > > Caveat: it's depending on https://github.com/ChorusOne/lighthouse fork of sigp/lighthouse that need to have this PR merged : https://github.com/sigp/lighthouse/pull/7145/files
> > 
> > 
> > Once the lighthouse pr is merged, please bump the `eth2_network_config` dependency. Please update test cases to verify the new network support (consider replacing instances of `holesky` with `hoodi`)
> 
> hoodi won't replace holesky, so fine to just keep it as an additional network.
> 
> i see the lighthouse pr was merged btw https://github.com/sigp/lighthouse/pull/7145/files

The updated dependency induced missing / incorrect trait implementations/imports, method availability mismatches, version conflicts, and upstream changes in Lighthouse.
It's still WIP, will fix the compilation errors from this WIP commit asap.

gwenall commented:
Support for Hoodi testnet and remove Goerli/Prater

This change adds support for the Hoodi testnet to the eth-staking-smith tool while removing the deprecated Goerli/Prater networks to align with Lighthouse changes. 
The implementation includes: 
- Added Hoodi testnet to the list of supported networks 
- Added the Hoodi fork_version to the network configuration 
- Removed Goerli/Prater networks which are no longer supported by Lighthouse 
- Updated tests to use Sepolia signatures instead of Goerli (I picked the closest testnet) 
- Fixed an issue with validator_seed_index handling in presigned_exit_message 
- Refactored test DepositDataJson to accommodate changes

kamsz commented:
hey @gwenall, could you please run cargo fmt? :)

mariavarvaroi commented on src/bls_to_execution_change/operations.rs:
```diff
@@ -66,10 +66,12 @@ impl SignedBlsToExecutionChangeValidator for SignedBlsToExecutionChange {
         );
 
         // verify signature
+        // Clone the genesis_validators_root to match expected type
```
This change seems unnecessary, you can just use `*genesis_validators_root`

mariavarvaroi commented on src/bls_to_execution_change/test.rs:
```diff
@@ -20,7 +20,7 @@ fn it_generates_signed_bls_to_execution_change() {
         let withdrawal_creds = utils::get_withdrawal_credentials(&withdrawal_pk.into(), 0);
         let credentials_hash = Hash256::from_slice(&withdrawal_creds);
-        hex::encode(&credentials_hash.as_bytes())
+        hex::encode(&credentials_hash.as_slice())
```
Just curious about the motivation for changing from `as_bytes()` to `as_slice()` - they seem equivalent to me

gwenall replied:
> Just curious about the motivation for changing from `as_bytes()` to `as_slice()` - they seem equivalent to me

I actually got this error when trying to use .as_bytes() on the type from alloy_primitives:

```
error[E0599]: no method named as_bytes found for struct alloy_primitives::bits::fixed::FixedBytes<32> in the current scope
   --> src/key_material.rs:132:39
    |
132 |         hex::encode(&credentials_hash.as_bytes())
    |                                       ^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
```

Where, as_bytes() isn’t available for FixedBytes<N>.
But I found that [as_slice() is the supported method](
https://docs.rs/alloy-primitives/latest/alloy_primitives/bits/fixed/struct.FixedBytes.html) for accessing the underlying bytes.

mariavarvaroi commented on src/key_material.rs:
```diff
@@ -129,7 +129,7 @@ mod test {
         let withdrawal_creds = utils::get_withdrawal_credentials(&withdrawal_pk.into(), 0);
         let credentials_hash = Hash256::from_slice(&withdrawal_creds);
-        hex::encode(&credentials_hash.as_bytes())
+        hex::encode(&credentials_hash.as_slice())
```
This might be unnecessary

gwenall replied:
I hit the [same issue here ](https://github.com/ChorusOne/eth-staking-smith/pull/62/files#r2059113790)

gwenall commented on src/utils.rs:
```diff
@@ -31,7 +30,18 @@ pub(crate) fn scrypt() -> Kdf {
 ///
 /// Used for submitting deposits to the Eth1 deposit contract.
 pub(crate) fn get_withdrawal_credentials(pubkey: &PublicKeyBytes, prefix_byte: u8) -> Vec<u8> {
-    let hashed = ethereum_hashing::hash(&pubkey.as_ssz_bytes());
```
That'd be much better! Thanks! :-)

gwenall commented:
> hey @gwenall, could you please run cargo fmt? :)

Thanks, I just did :pray:

gwenall commented:
Hi @mariavarvaroi , @kamsz , and @jenpaff , just a gentle ping, to see whether the latest changes are ok for you.
Thanks

mikalsande commented:
@gwenall We have fixed a few issues in 603434086c5aa3275ab2d677492e019923b8984e, can you please rebase on main?

gwenall commented:
@mikalsande , It is now rebased. 
@mariavarvaroi , I fixed the clippy errors
Thanks

mikalsande commented on Cargo.toml:
```diff
@@ -36,7 +36,7 @@ tiny-bip39 = "1.0.0"
 # This must be pinned to a version that lighthouse uses
 tree_hash = "0.5.2"
 tree_hash_derive = "0.5.2"
```
https://github.com/ChorusOne/lighthouse/blob/664a8fae9650ecba457fccbb561c5c0d6c270e3b/Cargo.toml#L216-L217

Revision 664a8f of Lighthouse uses newer versions. Can you please update them.
```suggestion
tree_hash = "0.8"
tree_hash_derive = "0.8"
```

mikalsande reviewed (changes requested):
The tree_hash crates need to be updated to the same version that is used in Lighthouse.

mariavarvaroi reviewed (approved):
LGTM!

mikalsande reviewed (approved):
LGTM! The testsuite passed 👍
